controller/volume_controller.go

```diff
@@ -1456,7 +1456,23 @@ func (c *VolumeController) requestRemountIfFileSystemReadOnly(v *longhorn.Volume
 	if v.Status.State == longhorn.VolumeStateAttached && e.Status.CurrentState == longhorn.InstanceStateRunning {
 		fileSystemReadOnlyCondition := types.GetCondition(e.Status.Conditions, imtypes.EngineConditionFilesystemReadOnly)
 
-		if fileSystemReadOnlyCondition.Status == longhorn.ConditionStatusTrue {
+		isPVMountOptionReadOnly := false
+		kubeStatus := v.Status.KubernetesStatus
+		if kubeStatus.PVName != "" {
+			pv, err := c.ds.GetPersistentVolume(kubeStatus.PVName)
+			if err != nil {
+				log.Warnf("Failed to get PV when checking the mount option of the volume")
+				return
+			}
+			for _, opt := range pv.Spec.MountOptions {
+				if opt == "ro" {
+					isPVMountOptionReadOnly = true
+					break
+				}
+			}
+		}
+
+		if fileSystemReadOnlyCondition.Status == longhorn.ConditionStatusTrue && !isPVMountOptionReadOnly {
```

Comment:
When a mount point is ready-only, it can be:
- mount with `ro`
- volume engine is unresponsive or error

What we assumed is the second case, but how about that the mount point is `ro` and the second case happens, how to handle this mixed case?

Reply:
or do we try to check if the read operation is working? if yes, then we can differentiate that.

Reply:
volume engine crashing will be handled in other place
https://github.com/longhorn/longhorn-manager/blob/e355c6364a2abbd30b3652fbf47f1ad0ab113797/controller/volume_controller.go#L1398-L1424
For that case we will request remount no matter what, read-only is only a side effect of engine crashing

Here we catch other unexpected "ro" cases

Reply:
you can check my HackWeek Note for how we handle engine crash
https://docs.google.com/presentation/d/1aFtPX5yY53TGbndy3D2IRs5SrBxZigBRCiuLt3fMaW4/edit#slide=id.g26073da28eb_0_530